File: java/src/main/java/com/penapereira/patterns/mediator/ConcreteMediator.java
package com.penapereira.patterns.mediator;

/** Knows its colleagues and implements the cooperative behaviour: route to the other one. */
public final class ConcreteMediator implements Mediator {

    private Colleague colleague1;
    private Colleague colleague2;

    public void setColleague1(Colleague colleague) {
        this.colleague1 = colleague;
    }

    public void setColleague2(Colleague colleague) {
        this.colleague2 = colleague;
    }

    @Override
    public void notify(Colleague sender, String message) {
        Colleague target = sender == colleague1 ? colleague2 : colleague1;
        target.receive(message);
    }
}

File: typescript/src/mediator/concrete-mediator.ts
import type { Colleague } from "./colleague.ts";
import type { Mediator } from "./mediator.ts";

/** Knows its colleagues and implements the cooperative behaviour: route to the other one. */
export class ConcreteMediator implements Mediator {
  private colleague1: Colleague | undefined;
  private colleague2: Colleague | undefined;

  setColleague1(colleague: Colleague): void {
    this.colleague1 = colleague;
  }

  setColleague2(colleague: Colleague): void {
    this.colleague2 = colleague;
  }

  notify(sender: Colleague, message: string): void {
    const target = sender === this.colleague1 ? this.colleague2 : this.colleague1;
    target?.receive(message);
  }
}
